controllers/reports.js

```diff
@@ -0,0 +1,142 @@
+import models from '../models';
+
+const { User, Report, article } = models;
+
+const reporting = {
+  /**
+   * Authenticated user should be able to report
+   * an article that violate terms of aggrement.
+   *
+   * @param {*} req request object.
+   * @param {*} res response object.
+   * @returns {*}   response to be returned.
+   */
+  reportArticle: async (req, res) => {
+    const { slug } = req.params;
+    const { message } = req.body;
+    const { id: reporter } = req.user;
+    try {
+      const findArticle = await article.findOne({
+        where: { slug },
+        attributes: ['id', 'slug', 'title', 'body']
+      });
+      if (!findArticle) {
+        return res.status(404).send({
+          status: res.statusCode,
+          error: 'This article is not found',
```

Comment:
@niyobobo I think this error message should be specific. We have a lot of "article not found" in our code. I think you could say this. "The article you're trying to report is not found"

Reply:
> @niyobobo I'm not talking about getting one article or many! I'm talking about **how the API URL is written.** I was suggesting that the URL should be `URL: /api/v1/report/articles/:slug` for reporting an article or deleting one. As we are using the plural when referring to objects i.e `POST /api/v1/articles not /api/v1/article` for creating an article.

@avpaul  Changes applied

Reply:
@aaronsekisambu changes applied